contracts/eosio.system/src/voting.cpp

```diff
@@ -371,6 +406,9 @@ namespace eosiosystem {
                auto& d = producer_deltas[p];
                d.first -= voter->last_vote_weight;
                d.second = false;
+               if (self_stake_boost && p == voter_name) {
+                  d.first -= 10*voter->last_vote_weight;
```

Comment:
We should make the 10x multiplier configurable in the voting singleton so it can be adjusted later.